code/modules/clothing/clothing.dm

```diff
@@ -109,8 +109,7 @@
 
 /// Set the clothing's integrity back to 100%, remove all damage to bodyparts, and generally fix it up
 /obj/item/clothing/proc/repair(mob/user, params)
-	damaged_clothes = CLOTHING_PRISTINE
-	update_clothes_damaged_state()

```

Comment:
the issue was that this wasn't being passed the repaired var, resulting in the item remaining damaged.